Sources/SwerverCore/Response/HttpResponse.swift

```diff
@@ -3,26 +3,23 @@ import Foundation
 public final class HttpResponse {
 
     public let statusCode: String
-    public let contentType: String
-    public let dateTime: Date = Date()
     public let body: String
-    public let contentLength : Int
-    public let allowedMethods: [String]
+    public let headers: [ResponseHeader:String]
```

Comment:
What is `ResponseHeader`?

Reply:
ResponseHeader is an enum that replaces the need to hardcode each header such as "Content-Type", "Allow"

Reply:
Can headers be arbitrary or is there a precise list of available headers (in the RFC)? If they can be arbitrary, then making it an enum is fairly limiting.

#futurepullrequest

Reply:
Can you clarify what you mean by error prone? If I am allowed to use any string as a header, what errors is this preventing?

Personally, I would just get rid of the header enum, but if you like the utility it provides, this might be a good compromise: You could provide an enum for a list of standard headers, but update `set(header:)` to take a `String`. e.g. `set(header: HttpHeader.allow.rawValue, to: "GET,HEAD")`

Reply:
Without the enum you would have to do: set(header: "header", to: "GET, HEAD") which can be inconsistent from response to response. Furthermore, it's error-prone because the "header" must now be matched exactly in order to retrieve the desired field and could return nil in the case of get(header: "Header" ...). it's also tedious to track down what and where the problem is in a situation like this.